File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        for (auto i = 0; i < nums.size(); i++) {
            int second = target - nums[i];
            if (mp.find(second) != mp.end()) return {mp[second],i};
            mp[nums[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int curr = nums[i];
            int find = target - curr;
            if (map.containsKey(find))
                return new int[] {map.get(find), i};
            map.put(curr, i);
        }
        return new int[] {};
    }
}